SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunnerTests/OptionTests.swift

```diff
@@ -211,5 +211,10 @@ class OptionTests: XCTestCase {
         XCTAssertEqual(reflectedSome!.field, 123)
         XCTAssertNil(reflectedNone)
     }
+    
+    func testFailableInitializer() {
+        XCTAssertEqual(FailableInitType(false), nil)
+        XCTAssertNotEqual(FailableInitType(true), nil)
```

Comment:
Let's make sure that we can use the type. This gives us more confidence that we didn't, say, bridge `Some(bool)` instead of `Some(FailableInitType)`.

Maybe
```
let some = FailableInitType(true)
XCTAssertEqual(some!.count(), 123);
```